### DISCUSSION TITLE
What is the best way to make changes to theme after deployment?

### DISCUSSION THREAD

**Discussion by SamuelAshbrook** [gregsullivan/_tw#147] [General]
Hi. Sorry if these sounds like a stupid question but I was just wondering what the best steps are to make changes to a theme after it has been bundled and deployed to production.

At the moment I am keeping the development build to make my changes and then overwriting the old theme with the new theme however, I feel there must be a more streamlined approach?

Thank you.

**Comment by gregsullivan** [comment]
Hi Samuel—Thanks for posting this!

It sounds like you're asking about approaches to deployment. Is that right? There is a [documentation page covering deployment](https://underscoretw.com/docs/deployment/) that might be a good starting point if you haven't read it already.

Uploading the zip file generated by `npm run bundle` via Appearance > Themes > Add New Theme is a viable deployment method, especially if you're not deploying that often.

In terms of other deployment methods, there are lots of options for deploying via command-line scripts or as part of GitHub Actions. It's hard to provide a single recipe for these approaches, but the documentation page linked above outlines what you need to keep in mind. If you're creating your own development workflow, you'll also need to remember that `npm run bundle` does asset versioning for JavaScript and CSS files, so you'll need to take care of that as well.

If I misunderstood your question, or if you have follow-up questions, please let me know!
